docs/arc42.md

```diff
@@ -0,0 +1,725 @@
+    end
+    alt Recipient Registers Key
+        Recipient->>Mediator: Keylist Update
+        Mediator-->>Recipient: Keylist Update Response
+    end
+```
+
+**Illustration: Runtime Interaction**
+
+1. **Mediate Request**: The recipient sends a `mediate-request` message to the mediator, initiating the mediation process.
+
+2. **Mediator's Response**:
+   - **Grant**: If the mediator agrees, it responds with a `mediate-grant` message, providing necessary routing details.
+   - **Deny**: If the mediator declines, it sends a `mediate-deny` message.
+
+3. **Key Registration**: Upon receiving a grant, the recipient registers its keys with the mediator using a `keylist-update` message. The mediator acknowledges with a `keylist-update-response`.
+
+This interaction ensures that messages intended for the recipient are appropriately routed through the mediator, enhancing communication reliability and security.  
+
+---
+
+## 7. Deployment View
+
+### 7.1 Deployment Strategy
+
+The deployment of the `didcomm-mediator-rs` project leverages a scalable, distributed architecture designed for high availability, fault tolerance, and efficient message routing.
+
+---
+
+### 7.2 Deployment Models
+
+#### 1. **Single Instance Deployment (Minimal Setup)**
+
+**Use Case**: Suitable for development or testing environments.
+
+- A single mediator instance handles all requests and routing.
+- Simplifies deployment but lacks fault tolerance and scalability.
+
+**Advantages**:
+- Minimal resource usage.
+- Easy to configure and maintain.
+
+**Disadvantages**:
+- No fault tolerance: if the mediator fails, the service becomes unavailable.
+- Limited scalability.
+
+---
+
+#### 2. **Distributed Deployment with Load Balancer (Recommended)**
+
+**Use Case**: Ideal for production environments requiring high availability and scalability.
+
+- Multiple mediator instances are deployed behind a load balancer.
+- The load balancer distributes requests evenly across mediators, ensuring efficient utilization of resources.
+- Supports horizontal scaling by adding more mediator instances as needed.
+
+**Advantages**:
+- High availability: If one mediator instance fails, others continue handling requests.
+- Scalable: New instances can be added to handle increased load.
+- Efficient routing: Load balancer optimizes resource utilization.
+
+**Disadvantages**:
+- Slightly higher complexity due to additional components (load balancer).
+
+---
+
+### 7.3 Deployment Diagram
+
+Below is a component diagram showcasing the deployment with multiple mediators and a load balancer:
+
+```mermaid
+graph TD
+    subgraph Internet
+        User1[Recipient Agent]
+        User2[Sender Agent]
+    end
+
+    User1 --> LB[Load Balancer]
+    User2 --> LB
+
+    subgraph Mediator Cluster
+        Mediator1[Mediator Instance 1]
+        Mediator2[Mediator Instance 2]
+    end
+
+    LB --> Mediator1
+    LB --> Mediator2
+
+    Mediator1 --> DB1[Message Storage DB]
+    Mediator2 --> DB1
+```
+
+---
+
```

Comment:
instead of this diagram use this instead 
https://drive.google.com/file/d/1EUW6UqAAo5YsR073vdLrj1DWVnYix5MV/view